feat: Support exporting metrics in Prometheus format

chaneytech opened the pull request:
Support exporting metrics in Prometheus format，set OTEL_METRICS_EXPORTER=prometheus.

config prometheus addr by OTEL_EXPORTER_PROMETHEUS_HOST and OTEL_EXPORTER_PROMETHEUS_PORT，see https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#prometheus-exporter


Issue：https://github.com/alibaba/opentelemetry-go-auto-instrumentation/issues/254

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/alibaba/opentelemetry-go-auto-instrumentation?pullRequest=255) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/alibaba/opentelemetry-go-auto-instrumentation?pullRequest=255) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/alibaba/opentelemetry-go-auto-instrumentation?pullRequest=255) it.</sub>

123liuziming commented:
Could you please add a demo for using prometheus exporter? Thanks!

chaneytech commented:
> Could you please add a demo for using prometheus exporter? Thanks!

1. add environment variables
OTEL_METRICS_EXPORTER=prometheus
OTEL_EXPORTER_PROMETHEUS_HOST=0.0.0.0
OTEL_EXPORTER_PROMETHEUS_PORT=9464
2. otel go build app
3. When the application starts, use curl localhost:9464/metrics to check the metrics data. If the response is successful and returns the expected metrics, it indicates that the application has started successfully and the metrics service is working properly.

123liuziming commented:
> > Could you please add a demo for using prometheus exporter? Thanks!
> 
> 1. add environment variables
>    OTEL_METRICS_EXPORTER=prometheus
>    OTEL_EXPORTER_PROMETHEUS_HOST=0.0.0.0
>    OTEL_EXPORTER_PROMETHEUS_PORT=9464
> 2. otel go build app
> 3. When the application starts, use curl localhost:9464/metrics to check the metrics data. If the response is successful and returns the expected metrics, it indicates that the application has started successfully and the metrics service is working properly.

Cool, Could you please put theses into` https://github.com/alibaba/opentelemetry-go-auto-instrumentation/tree/main/example` directory? And the CI seems failed, so you may need to fix the CI first.

chaneytech commented:
> > > Could you please add a demo for using prometheus exporter? Thanks!
> > 
> > 
> > 
> > 1. add environment variables
> >    OTEL_METRICS_EXPORTER=prometheus
> >    OTEL_EXPORTER_PROMETHEUS_HOST=0.0.0.0
> >    OTEL_EXPORTER_PROMETHEUS_PORT=9464
> > 2. otel go build app
> > 3. When the application starts, use curl localhost:9464/metrics to check the metrics data. If the response is successful and returns the expected metrics, it indicates that the application has started successfully and the metrics service is working properly.
> 
> Cool, Could you please put theses into` https://github.com/alibaba/opentelemetry-go-auto-instrumentation/tree/main/example` directory? And the CI seems failed, so you may need to fix the CI first.

ok.

chaneytech commented:
I have added two subdirectories under example/metrics: 
* example/metrics/otlp-exporter
*  example/metrics/prometheus-exporter. 

The original source code from example/metrics has been moved to example/metrics/otlp-exporter. 
In example/metrics/prometheus-exporter, I have added a demo for Prometheus metrics. Which can retrieve metric data from localhost:9464/metrics. Additionally, the Prometheus server is accessible at localhost:9090 for viewing the metrics.

The PR is based on the example provided in the OpenTelemetry Go Contrib repository: https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/examples/prometheus/main.go.

123liuziming commented:
> I have added two subdirectories under example/metrics:
> 
> * example/metrics/otlp-exporter
> * example/metrics/prometheus-exporter.
> 
> The original source code from example/metrics has been moved to example/metrics/otlp-exporter. In example/metrics/prometheus-exporter, I have added a demo for Prometheus metrics. Which can retrieve metric data from localhost:9464/metrics. Additionally, the Prometheus server is accessible at localhost:9090 for viewing the metrics.
> 
> The PR is based on the example provided in the OpenTelemetry Go Contrib repository: https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/examples/prometheus/main.go.

Excellent! I will check it later, thanks.

chaneytech commented:
@123liuziming Please check it again, thanks!

123liuziming reviewed (approved):
LGTM
